Tasks/VSMobileCenterUpload/vsmobilecenterupload.ts

```diff
@@ -0,0 +1,348 @@
+    };
+
+    request.patch({ url: publishPackageUrl, headers: headers, json: publishBody }, (err, res, body) => {
+        responseHandler(defer, err, res, body, () => {
+            defer.resolve();
+        });
+    })
+
+    return defer.promise;
+}
+
+/**
+ * If the input is a single file, upload this file without any processing.
+ * If the input is a dSYM folder, zip the parent and upload the zip so dSYM folder appears on the root of the archive
+ * If the input is a folder, zip the input folder so all files under this folder appears on the root of the archive 
+ */
+function prepareSymbols(symbolsPath: string, packParentFolder: boolean): Q.Promise<string> {
+    tl.debug("-- Prepare symbols")
+    let defer = Q.defer<string>();
+  
+    let stat = fs.statSync(symbolsPath);
+    if (stat.isFile() && !packParentFolder) {
+        // single file - Android source mapping txt file
+        tl.debug(`---- symbol file: ${symbolsPath}`)
+        defer.resolve(symbolsPath);
+    } else { 
+        if (packParentFolder) {
+            tl.debug(`---- Take the parent folder of ${symbolsPath}`);
+            symbolsPath = path.dirname(symbolsPath); 
+        }
+
+        tl.debug(`---- Creating symbols from ${symbolsPath}`);
+        let zipStream = utils.createZipStream(symbolsPath, utils.isDsym(symbolsPath));
+        let workDir = tl.getVariable("System.DefaultWorkingDirectory");
+        let zipName = path.join(workDir, `${path.basename(symbolsPath)}.zip`); 
+        utils.createZipFile(zipStream, zipName).
+            then(() => {
+                tl.debug(`---- symbol file: ${zipName}`)
+                defer.resolve(zipName);
+            });
+    }
+
+    return defer.promise;
+}
+
+function beginSymbolUpload(apiServer: string, apiVersion: string, appSlug: string, symbol_type: string, token: string): Q.Promise<SymbolsUploadInfo> {
+    tl.debug("-- Begin symbols upload")
+    let defer = Q.defer<SymbolsUploadInfo>();
+
+    let beginSymbolUploadUrl: string = `${apiServer}/${apiVersion}/apps/${appSlug}/symbol_uploads`;
+    tl.debug(`---- url: ${beginSymbolUploadUrl}`);
+
+    let headers = {
+        "X-API-Token": token
+    };
+
+    let symbolsUploadBody = { "symbol_type": symbol_type };
+    
+    request.post({ url: beginSymbolUploadUrl, headers: headers, json: symbolsUploadBody}, (err, res, body) => {
+        responseHandler(defer, err, res, body, () => {
+            let symbolsUploadInfo: SymbolsUploadInfo = {
+                symbol_upload_id: body['symbol_upload_id'],
+                upload_url: body['upload_url'],
+                expiration_date: body['expiration_date']
+            }
+
+            defer.resolve(symbolsUploadInfo);
+        });
+    })
+
+    return defer.promise;
+}
+
+function uploadSymbols(uploadUrl: string, file: string): Q.Promise<void> {
+    tl.debug("-- Uploading symbols...");
+    let defer = Q.defer<void>();
+    tl.debug(`---- url: ${uploadUrl}`);
+
+    let stat = fs.statSync(file);
```

Comment:
The symbol file is guaranteed to be present here since this is after "prepareSymbols" call. 

In the new framework, everything can be mocked so it's not necessarily to have task-lib be the middle man anymore.